### ISSUE TITLE
merge ranks an arm with zero seed overlap against a present control without any paired_vs_control block

### ISSUE THREAD

**Issue by swissyai** [SlopDotCash/asi#49]
## Problem

`merge_shards` silently publishes a ranked, schema-valid summary entry for an arm that shares **zero seeds** with a present control. The pairing block is skipped by the `and common` guard:

```python
common = [s for s in seeds if s in control]
if name != control_name and common:      # empty intersection -> no block, no error
    ...
    entry["paired_vs_control"] = {...}
```

The entry still ranks by raw `average_online_accuracy_mean` — an unpaired, cross-seed comparison presented alongside paired ones, with nothing in the artifact marking it as unpaired. Every comparison rule in this campaign requires shared seeds ("identical seeds across arms => paired comparison", RUNBOOK) and the repo treats silent ambiguity in the measurement path as a defect (#41, #44, #46). This is the zero-overlap case #44/#45 explicitly left out of scope.

Reachable through the standard `merge` CLI by an incomplete wave or a seed-list typo — no unusual flags needed.

## Repro (at `main` `a0e3412`)

Control shards on seeds 0-1, candidate `upgd_l2init` on seed 2 only (tiny protocol config, same shapes as the test suite's):

```text
control_name: upgd_w_control
rank entry: upgd_l2init      seeds=[2]    mean=0.2556 paired_vs_control=ABSENT
rank entry: upgd_w_control   seeds=[0, 1] mean=0.2556 paired_vs_control=ABSENT
```

The unpaired arm ranks first; a reader of the summary has no signal that its position is not a paired result. Full script below.

<details><summary>overlap_repro.py</summary>

```python
import json
import tempfile
from pathlib import Path

import numpy as np
import jax.numpy as jnp
import jax.random as jr

from alberta_framework.benchmarks.ipmnist_screening import (
    merge_shards,
    run_screening_config,
    screening_spec,
    shard_payload,
)
from alberta_framework.benchmarks.upgd_ipmnist import IPMNISTConfig

SMALL = IPMNISTConfig(
    n_tasks=3, task_length=30, input_dim=12, hidden1=8, hidden2=6, n_classes=5
)
key = jr.key(1234)
kx, ky = jr.split(key)
x = np.asarray(jr.uniform(kx, (64, SMALL.input_dim), jnp.float32, -1.0, 1.0))
y = np.asarray(jr.randint(ky, (64,), 0, SMALL.n_classes))

with tempfile.TemporaryDirectory() as td:
    paths = []
    for name, seed in (("upgd_w_control", 0), ("upgd_w_control", 1), ("upgd_l2init", 2)):
        result = run_screening_config(x, y, screening_spec(name), seed=seed, config=SMALL)
        p = Path(td) / f"{name}_seed{seed}.json"
        p.write_text(json.dumps(shard_payload(result)), encoding="utf-8")
        paths.append(p)
    summary = merge_shards(paths, control_name="upgd_w_control", slope_window=2)

print("control_name:", summary["control_name"])
for entry in summary["results"]:
    print(
        f"rank entry: {entry['config_name']:16s} seeds={entry['seeds']} "
        f"mean={entry['average_online_accuracy_mean']:.4f} "
        f"paired_vs_control={'present' if 'paired_vs_control' in entry else 'ABSENT'}"
    )
```

</details>

## Historical safety of a refusal

All 8 pinned `outputs/ipmnist_screening/summary*.json` verified: every non-control entry carries `paired_vs_control` with `seeds` equal to its own full seed list — no unpaired entries, no partial intersections. A deterministic refusal on zero overlap therefore breaks no sanctioned historical merge, same situation #45 established for the absent-control case.

## Proposed fix

Raise a deterministic `ValueError` naming the arm, its seeds, and the control's seeds when the intersection is empty. **Partial** overlap stays allowed — it pairs on the intersection and the block records those seeds visibly, which is legitimate and historically unused but harmless. Failing-test-first, plus a companion pin that partial overlap keeps merging on the intersection so the refusal cannot over-reach.

Composes with #45: its absent-control refusal sits before the per-arm loop, this one inside it; the hunks are independent.

I'm taking this one — code-only PR to follow shortly from this account.

AI provider/model: anthropic / claude-fable-5. Client: claude-code.
